x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx

```diff
@@ -34,137 +29,41 @@ function EmbeddedMapComponent({
   filters: Filter[];
   dataView?: DataView;
```

Comment:
The `dataView` passed into this component is an ad-hoc data view created from the `useAdHocApmDataView` hook. However, the ad-hoc dataview is [created without specifying a `timeFieldName`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/public/hooks/use_adhoc_apm_data_view.ts#L32). So Maps does not think the source is time aware and will not respond to changes to the time picker. This is a bug that currently exists in `main` as well. 

The `useAdHocApmDataView` hook is used in multiple places in this plugin so I did not feel comfortable making any changes there.